src/core/qgsogcutils.cpp

```diff
@@ -312,10 +321,29 @@ QgsGeometry QgsOgcUtils::geometryFromGMLPolygon( const QDomElement &geometryElem
     {
       return QgsGeometry();
     }
-    if ( readGMLPositions( exteriorPointList, posElement ) != 0 )
+    if ( readGMLPositions( exteriorPointList, posElement ) != false )
     {
       return QgsGeometry();
     }
+
+    if ( geomSrs.isValid() )
+    {
+      QgsCoordinateReferenceSystem exteriorSrs;
+      const QDomElement ringElement {  exteriorList.at( 0 ).firstChild().toElement() };
+      if ( ringElement.hasAttribute( QStringLiteral( "srsName" ) ) )
+      {
+        exteriorSrs.createFromString( ringElement.attribute( QStringLiteral( "srsName" ) ) );
+      }
+      if ( exteriorSrs != geomSrs )
+      {
+        const QgsCoordinateTransform transformer { exteriorSrs, geomSrs, QgsProject::instance()->transformContext() };
```

Comment:
Using QgsProject::instance() is definitely not server friendly here. You'll need to instead add a transform context argument to the method and use that instead (or better, add a "GML Conversion Context" struct/class containing the transform context member)

Reply:
yeah, I know. I'll do if we decide to go further with this PR (see comments below)